test/jdk/java/awt/im/4959409/bug4959409.java

```diff
@@ -21,52 +21,139 @@
+        CountDownLatch frameVisibleLatch = new CountDownLatch(1);
+        CountDownLatch keyPressedEventLatch = new CountDownLatch(1);
+        final Point[] points = new Point[1];
+        final Rectangle[] rect = new Rectangle[1];
+
+        SwingUtilities.invokeAndWait(() -> {
+            frame = new JFrame("Test bug4959409");
+            jTextField = new JTextField();
+            jLabel = new JLabel();
+            frame.setLayout(new BorderLayout());
+            frame.addWindowListener(new WindowAdapter() {
+                @Override
+                public void windowOpened(WindowEvent e) {
+                    super.windowOpened(e);
+                    frameVisibleLatch.countDown();
+                    System.out.println("Frame opened");
+                }
+            });
+
+            jTextField.addKeyListener(new KeyAdapter() {
+
+                @Override
+                public void keyPressed(KeyEvent keyEvent) {
+                    super.keyPressed(keyEvent);
+                    int code = keyEvent.getKeyCode();
+                    int mod = keyEvent.getModifiersEx();
+                    if (code == '1' && mod == KeyEvent.SHIFT_DOWN_MASK) {
+                        keyPressedEventLatch.countDown();
+                        jLabel.setText("keyPressed received for Shift+1");
+                        System.out.println("keyPressed received for Shift+1");
+                    } else {
+                        jLabel.setText("Did not received keyPressed for Shift+1");
+                        System.out.println("Did not received keyPressed for Shift+1");
+                    }
+                }
+            });
 
-    public void keyPressed(KeyEvent e) {
-        int code = e.getKeyCode();
-        int mods = e.getModifiers();
-        if (code == '1' && mods == KeyEvent.SHIFT_MASK) {
-            label.setText("KEYPRESS received for Shift+1");
-        } else {
-            label.setText(" ");
+            Container container = frame.getContentPane();
+            container.add(jTextField, BorderLayout.SOUTH);
+            container.add(jLabel, BorderLayout.CENTER);
+            frame.setSize(300, 300);
+            frame.setLocationRelativeTo(null);
+            frame.setAlwaysOnTop(true);
+            frame.setDefaultCloseOperation(JFrame.EXIT_ON_CLOSE);
+            frame.setVisible(true);
+        });
+
+        Robot robot = new Robot();
+        robot.setAutoDelay(DELAY);
+        robot.waitForIdle();
+
+        if (!frameVisibleLatch.await(TIMEOUT, TimeUnit.SECONDS)) {
+            throw new RuntimeException("Frame is not visible after " + TIMEOUT + "  seconds");
+        }
+
+        SwingUtilities.invokeAndWait(() -> {
+            points[0] = jTextField.getLocationOnScreen();
+            rect[0] = jTextField.getBounds();
+        });
+
+        clickTextField(robot, points[0].x + rect[0].width / 2, points[0].y + rect[0].height / 2);
```

Comment:
I'd probably wrap this line, it doesn't fit even in 120 columns.

Reply:
done